instrumentation/kafka-clients/src/main/java/brave/kafka/clients/TracingProducer.java

```diff
@@ -59,10 +62,17 @@
     return this.send(record, null);
   }
 
-  /** We wrap the send method to add tracing. */
-  @Override
-  public Future<RecordMetadata> send(ProducerRecord<K, V> record, @Nullable Callback callback) {
-    Span span = tracing.tracer().nextSpan();
+  /**
+   * We wrap the send method to add tracing.
+   */
+  @Override public Future<RecordMetadata> send(ProducerRecord<K, V> record, @Nullable Callback callback) {
+    final Span span;
+    if (tracing.tracer().currentSpan() == null) {
```

Comment:
This style assumes someone didn't use the apis that we tell them to use.. probably needs more comment here (and tests if we decide to go with it) explaining why we are checking for someone not using things.

At any rate this will hurt perf.. if you want to hurt less, you can do this:

```java
TraceContext maybeParent = tracing.currentTraceContext().get();
if (maybeParent == null) { // check to see if someone forgot to call something
 ... calling extract will also hurt perf! that's why we should be careful not to call this unless we have good reason commented
} else {
... tracer.newChild(parent);
}
```

Reply:
thanks for your comments @adriancole 
Will add comments and test for this. Also, we can hurt a bit less if we only perform `extract` when there are headers available, then users that are not using them will not be affected.